activities.el

```diff
@@ -809,9 +932,60 @@ activity's name is NAME."
   "Return an activity read with completion from ACTIVITIES.
 PROMPT is passed to `completing-read' by way of `format-prompt',
 which see, with DEFAULT."
-  (let* ((prompt (format-prompt prompt default))
-         (names (activities-names activities))
-         (name (completing-read prompt names nil t nil 'activities-completing-read-history default)))
+  (pcase-let*
+      ((names (activities-names activities))
+       (max-age (activities--oldest-age activities))
+       (`(,old-col ,new-col ,blend-frac) activities-annotation-colors)
+       (annotation-function
```

Comment:
FWIW, I'll probably move the annotation function into a `cl-labels` form; just a preference of mine.

Reply:
Honestly it does feel we are packing a _lot_ into this function body.  I'd certainly reach for an internal function to keep it clearer, but again what matters is what's clear to _you_.  I converted them both to named local functions with `cl-labels`.  I didn't think I'd like it, but I do like that it's explicit that they are functions now in the call to `completing-read`, and have docstrings for clarity.